### DISCUSSION TITLE
Feature Request: Ability to Customize Photo Interval Settings via QR Code (Resolution, Lens Type, Shutter Speed)

### DISCUSSION THREAD

**Discussion by m-h-williams** [gopro/labs#1105] [Feature Requests]
Hello GoPro Labs Team! 

We have been using the photo interval mode on our GoPro for various applications and have noticed that the image quality is significantly better than when using the timelapse mode. However, the timelapse mode has the added benefit of being customizable via QR code generation in GoPro Labs, allowing us to adjust settings such as resolution, lens type, and shutter speed.

We would love to see this same level of customization extended to the photo interval mode. Currently, the QR code generator does not support detailed customization of the photo interval settings like it does for timelapse. Being able to control settings such as resolution, lens type, and shutter speed in photo interval mode would greatly enhance its usability, especially for detailed work that relies on the higher image quality of this mode.

**Request:**

Add options to the GoPro Labs QR code generator to customize the following settings in photo interval mode:

- Resolution (e.g., 12MP, 20MP, etc.)
- Lens type (e.g., Wide, SuperView, Linear, Narrow, etc.)
- Shutter speed (e.g., 1/30, 1/60, 1/450, etc.)
- Provide the ability to set a more specific interval time, similar to what is available for timelapse, but with more control over the exposure and image quality parameters.

**Why This is Useful:**

- Image Quality: The images captured via photo interval mode are often higher quality than those in timelapse, which can be crucial for use cases that require detailed, high-resolution images.
- Customization Flexibility: Having the flexibility to adjust key settings such as resolution, lens type, and shutter speed would allow users to better tailor the photo interval mode for various environmental and shooting conditions, improving both the creative and technical outcome.
- Consistency Across Modes: Extending the same QR customization capabilities that are available for timelapse mode to photo interval mode would create a more consistent user experience for those who frequently switch between these modes.

We hope this feature can be considered for future updates to GoPro Labs, as it would greatly enhance the camera’s versatility and customization options.

Thank you for your consideration!

**Comment by dnewman-gpsw** [comment]
Labs mostly controls features that are either already in the menus, or in the camera, but not exposed in the menus.  Photo have Wide and Linear, but not Superview or narrow.  So Labs can't add those, particularly as those are so easily done in post.

Resolution, why do you want lower resolutions? Full res is the best option for post processing.

It can control the shutter more precisely using $EXPQ=x (where shutter = 1/x). You use this feature while the camera is set to Auto shutter.

For timelapse and interval photo you already have options 0.5s,1,2,3,5,10... etc. There are a lot to choose from.  If you want custom weird intervals like 18s or 25s, use you a Labs script to take a Photo with the interval needed.  

The JPEG compression level does change in timelapse vs single Photo mode, but I'm not seeing any compression quality difference between interval and time-lapse photo. This is simply to reduce the data-rate for the shorter intervals. If you used Photo compression quality during a short interval timelapse, you may crash the camera if the SD card doesn't keep up. This is still a Labs opportunity, as crashing the camera would be then under your control. ;) I will see about adding JPEG quality control.

You questions in general implies the interval and time-lapse photo, have significant feature differences, yet I'm not seeing that.

**Reply by m-h-williams** [reply]
Thanks for your response here. I appreciate you sharing the purpose behind labs and why certain features aren't available for certain modes in addition to the code for controlling the shutter speed. 

Would you be able to share the differences behind interval photo and time-lapse photo then? Why have these two options if the quality doesn't change?  A bit of background: 

- My understanding after looking online was that time-lapse photos are optimized for the stitching of images that will become a video, so the photos taken have the same (or similar) settings applied to every photo. This way when the photos are played back as frames of a video, there aren't jarring differences from frame-to-frame.

- Comparatively, interval photo uses the best settings for the environment without any regard for the previous or subsequent photo. So the settings can vary wildly from one photo to the next in an attempt to make each individual photo the best it can be.

- For our purposes, we are wanting to take images of the seafloor (1 still every 3 seconds) as we move along at a consistent rate of ~0.1m/s. Underwater photography has it's own set of challenges, as I'm betting you are aware, so we are using interval photo under the assumption that would be the best mode to capture high quality still images. We are capturing RAW images to give us the greatest flexibility post processing.  

I guess my request is really to include interval photo as a Photo-Mode in the GoPro QR Code Creator (like time-lapse) so we can customize a QR code that will include Pro-tune settings. We really appreciate the ease of having a QR code to show the GoPro that reduces user error and ensures the same settings are used again and between cameras.

**Reply by dnewman-gpsw** [reply]
True there are no QR codes yet for Interval Photo. As you are using Raw images, there is no issue with JPEG compression, you are using VC5 compression (much better.)  As for Interval vs time-lapse, I've been working with another underwater photogrammetry team, they are using time-lapse without issue (they just presented a paper : "Optimising GoPro Cameras for Underwater Photogrammetry" at AIMA Conference 2024.)  Time-lapse is just using weighted exposures from previous images, this is to prevent flicker in timelapse presentation. But the weighted exposure is not going to make any individual frames wrong, particularly from a slow moving ROV at 10cm/s.  

Paper: https://hive.curtin.edu.au/Research/GoProUP/

**Reply by m-h-williams** [reply]
That answers my question then, thank you! 

And thank you for sharing that link, I'm really excited to hear that another group has taken on the task of optimizing GoPros for underwater photogrammetry. We are very interested in building 3D models with our GoPro imagery, so knowing this work is being done by yourself and that team is very exciting. 

I don't appear to have access to the paper through that link, but I did submit my interest in becoming a beta-tester. We have super easy access to a dock, vessel, ROV, which makes it easy for us to test camera settings in a standardized way. I don't know where the breadth of this testing has been done, but we are in temperate waters off Seattle, WA, which can add its own complications (i.e. the water is green, dark, with reduced visibility) compared to clear tropical coral reefs. I hope we can help as testers!
